[BUG] Log message report NPE when using Jindo engine

**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c4368d]

goroutine 321 [running]:
github.com/fluid-cloudnative/fluid/api/v1alpha1.(*Dataset).GetObjectKind(0x0)
        <autogenerated>:1 +0x2d
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap.kubeObjectWrapper.MarshalLogObject({{0x243e710, 0x0}}, {0x2493c80, 0xc00047fcb0})
        /go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap/kube_helpers.go:67 +0x67
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*jsonEncoder).AppendObject(0xc00047fcb0, {0x7f1cfac80d60, 0xc00007a950})
        /go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/json_encoder.go:211 +0x74
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*jsonEncoder).AddObject(0xc00047fcb0, {0x22ace48, 0x7}, {0x7f1cfac80d60, 0xc00007a950})
        /go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/json_encoder.go:99 +0x5b
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.Field.AddTo({{0x22ace48, 0x7}, 0x2, 0x0, {0x0, 0x0}, {0x21128c0, 0xc00007a950}}, {0x2493c80, 0xc00047fcb0})
        /go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/field.go:121 +0x1c3
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.addFields({0x2493c80, 0xc00047fcb0}, {0xc001a0e100, 0x1, 0x1})
        /go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/field.go:210 +0x139
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.consoleEncoder.writeContext({0xc0017fdb60}, 0xc000afe6c0, {0xc001a0e100, 0x1, 0x1})
        /go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/console_encoder.go:145 +0x116
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.consoleEncoder.EncodeEntry({0xc0017fdb60}, {0x0, {0xc0e524caf7eab551, 0x2733d1e8aa, 0x329d860}, {0xc000175680, 0x15}, {0x22b6e52, 0x10}, {0x1, ...}, ...}, ...)
        /go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/console_encoder.go:119 +0x51e
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap.(*KubeAwareEncoder).EncodeEntry(0xc000778c78, {0x0, {0xc0e524caf7eab551, 0x2733d1e8aa, 0x329d860}, {0xc000175680, 0x15}, {0x22b6e52, 0x10}, {0x1, ...}, ...}, ...)
        /go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap/kube_helpers.go:126 +0x27c
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*ioCore).Write(0xc0017fdb90, {0x0, {0xc0e524caf7eab551, 0x2733d1e8aa, 0x329d860}, {0xc000175680, 0x15}, {0x22b6e52, 0x10}, {0x1, ...}, ...}, ...)
        /go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/core.go:86 +0xd9
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc000598000, {0xc001a0e100, 0x1, 0x1})
        /go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/entry.go:220 +0x33c
github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr.(*zapLogger).Info(0xc0017fdb30, 0x0, {0x22b6e52, 0x10}, {0xc00007e1c0, 0x2, 0x2})
        /go/src/github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr/zapr.go:198 +0xf7
github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/logr.Logger.Info({{0x246f588, 0xc0017fdb30}, 0x0}, {0x22b6e52, 0x10}, {0xc00007e1c0, 0x2, 0x2})
        /go/src/github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/logr/logr.go:249 +0x12b
github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx.(*JindoFSxEngine).shouldMountUFS(0xc0007d4790)
        /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx/ufs_internal.go:31 +0x245
github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx.(*JindoFSxEngine).PrepareUFS(0xc0007d4790)
        /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx/ufs.go:39 +0x7b
github.com/fluid-cloudnative/fluid/pkg/ddc/base.(*TemplateEngine).Setup(0xc0006f8a20, {{0x246ab38, 0xc001093f80}, {{0xc001a143f0, 0x7}, {0xc001a143e0, 0x10}}, {{0x246f588, 0xc001093fb0}, 0x0}, ...})
        /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/base/setup.go:65 +0x4ac
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileRuntime(0xc0002fe550, {0x247e210, 0xc0006f8a20}, {{0x246ab38, 0xc001093f80}, {{0xc001a143f0, 0x7}, {0xc001a143e0, 0x10}}, {{0x246f588, ...}, ...}, ...})
        /go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:243 +0x23c
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileInternal(0xc0002fe550, {{0x246ab38, 0xc001093f80}, {{0xc001a143f0, 0x7}, {0xc001a143e0, 0x10}}, {{0x
246f588, 0xc001093fb0}, 0x0}, ...})
        /go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:163 +0xef5
github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/jindo.(*RuntimeReconciler).Reconcile(0xc0001cb820, {0x246ab38, 0xc001093f80}, {{{0xc001a143f0, 0x7}, {0xc001a143e0, 0x10}
}})
```

**What you expect to happen:**
No NPE should be found.

**How to reproduce it**
It occurs when large datasets are being created concurrently. It's more likely to reproduce if you create datasets and jindoruntime with high frequency in a very short time.

**Additional Information**
```
func (e *JindoFSxEngine) shouldMountUFS() (should bool, err error) {
...
	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
	e.Log.Info("get dataset info", "dataset", dataset)
	if err != nil {
		return should, err
	}
...
```
The NPE may be caused by the `Log.Info` before checking `if err ! = nil`. A nil dataset will lead to zapcore's marshalling NPE.

--- a/pkg/ddc/alluxio/ufs_internal.go
+++ b/pkg/ddc/alluxio/ufs_internal.go
@@ -68,10 +68,10 @@ func (e *AlluxioEngine) totalFileNumsInternal() (fileCount int64, err error) {
 // shouldMountUFS checks if there's any UFS that need to be mounted
 func (e *AlluxioEngine) shouldMountUFS() (should bool, err error) {
 	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
-	e.Log.Info("get dataset info", "dataset", dataset)
 	if err != nil {
 		return should, err
 	}
+	e.Log.Info("get dataset info", "dataset", dataset)
 
 	podName, containerName := e.getMasterPodInfo()
 	fileUtils := operations.NewAlluxioFileUtils(podName, containerName, e.namespace, e.Log)

--- a/pkg/ddc/goosefs/ufs_internal.go
+++ b/pkg/ddc/goosefs/ufs_internal.go
@@ -64,10 +64,10 @@ func (e *GooseFSEngine) totalFileNumsInternal() (fileCount int64, err error) {
 // shouldMountUFS checks if there's any UFS that need to be mounted
 func (e *GooseFSEngine) shouldMountUFS() (should bool, err error) {
 	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
-	e.Log.Info("get dataset info", "dataset", dataset)
 	if err != nil {
 		return should, err
 	}
+	e.Log.Info("get dataset info", "dataset", dataset)
 
 	podName, containerName := e.getMasterPodInfo()
 	fileUtils := operations.NewGooseFSFileUtils(podName, containerName, e.namespace, e.Log)

--- a/pkg/ddc/jindofsx/ufs_internal.go
+++ b/pkg/ddc/jindofsx/ufs_internal.go
@@ -28,10 +28,10 @@ import (
 // shouldMountUFS checks if there's any UFS that need to be mounted
 func (e *JindoFSxEngine) shouldMountUFS() (should bool, err error) {
 	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
-	e.Log.Info("get dataset info", "dataset", dataset)
 	if err != nil {
 		return should, err
 	}
+	e.Log.Info("get dataset info", "dataset", dataset)
 
 	podName, containerName := e.getMasterPodInfo()
 	fileUtils := operations.NewJindoFileUtils(podName, containerName, e.namespace, e.Log)

